includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -559,11 +559,14 @@ private function process_node( DOMElement $node ) {
 			}
 		}
 
+		// Fix any values for width & height that may be AMP incompatible.
+		$this->adapt_dimensions_to_layout( $node );
```

Comment:
Rather than calling this method on every single node, what about adding an `AMP_Layout_Sanitizer` which is dedicated for the purpose of setting dimensions and layout? This sanitizer could do an XPath query for `//*[ @layout ]` to only get the elements that have the `layout` attribute to begin with. (This would also apply to querying for `data-amp-layout` attribute.)

Reply:
I recall also discussing moving this logic to `\AMP_Base_Sanitizer::set_layout()`.

In general, however, the `AMP_Tag_And_Attribute_Sanitizer` should be considered purely used for validation and sanitization. It is the last check to ensure valid AMP. So it can _remove_ elements and attributes, but in general it shouldn't be using to _add_ things to the DOM. This would be better suited for other sanitizers dedicated for that purpose.